src/resources/dart-3-migration.md

```diff
@@ -0,0 +1,512 @@
+but in Dart 3, you always get null safety, and it cannot be turned off.
+
+#### Scope
+
+This is an *unversioned* change, that applies to all Dart 3 code.
+
+#### Symptom
+
+Packages developed without null safety support will cause issues
+when resolving dependencies with `pub get`:
+
+```terminal
+$ dart pub get
+
+Because pkg1 doesn't support null safety, version solving failed.
+The lower bound of "sdk: '>=2.9.0 <3.0.0'" must be 2.12.0 or higher to enable null safety.
+```
+
+Libraries that opt out of null safety with [language version comments](/guides/language/evolution#per-library-language-version-selection)
+that select any language version below `2.12` will cause analysis or compilation
+errors:
+
+```terminal
+$ dart analyze .
+Analyzing ....                         0.6s
+
+  error • lib/pkg1.dart:1:1 • The language version must be >=2.12.0. 
+  Try removing the language version override and migrating the code.
+  • illegal_language_version_override
+```
+
+```terminal
+$ dart run bin/my_app.dart
+../pkg1/lib/pkg1.dart:1:1: Error: Library doesn't support null safety.
+// @dart=2.9
+^^^^^^^^^^^^
+```
+
+#### Migration
+
+A first step before beginning any migration to Dart 3, is to ensure your app or package has been 100% migrated to enable null safety. This must happen using a Dart `2.19` SDK, not a Dart 3 SDK. For details, see the [null safety migration guide](/null-safety/migration-guide).
+
+### Colon-syntax for default values
+
+For historical reasons, named optional parameters could specify their default
+value using either `:` or `=`. In Dart 3 only the `=` syntax is allowed.
+
+#### Scope
+
+This is an *unversioned* change, that applies to all Dart 3 code.
+
+#### Symptom
+
+Dart analysis produces errors like:
+
+```
+line 2 • Using a colon as a separator before a default value is no longer supported.
+```
+
+#### Migration
+
+Change from using colons:
+
+```dart
+int someInt({int x: 0}) => x;
+```
+
+To using equals:
+
+```dart
+int someInt({int x = 0}) => x;
+```
+
+This migration can be made manually, or automated with `dart fix`:
+
+```
+$ dart fix --apply --code=obsolete_colon_for_default_value
+```
+
+### `mixin`
+
+Pre-Dart 3, any `class` could be used as a `mixin`.
+In Dart 3, only types declared `mixin class` or `mixin`, may be used as a mixin.
+
+#### Scope
+
+This is a *versioned* change, that only applies to language version 3.0 or later.
+
+#### Migration
+
+Determine if the class is intended to be used as a mixin.
+
+If the class defines an interface, consider using `implements`.
+
+### `switch`
+
+Dart 3.0 interprets [switch cases](/language/control-flow#switch-and-case)
+as patterns [*TODO* LINK] instead of constant expressions. 
+
+#### Scope
+
+This is a *versioned* change, that only applies to language version 3.0 or later.
+
+#### Symptom
+
+Most constant expressions found in switch cases are valid patterns
+with the same meaning (named constants, literals, etc.).
+
+TODO: Give an example
```

Comment:
I wouldn't show an example here, except maybe the actual message from the `invalid_case_patterns` lint, to match the form of the other "Symptom" sections.